cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -215,68 +216,87 @@ runQueryTip (AnyConsensusModeParams cModeParams) network mOutFile = do
         eResult <- liftIO $ queryEraHistoryAndSystemStart localNodeConnInfo Nothing
         case eResult of
           Left acqFail -> left (ShelleyQueryCmdAcquireFailure acqFail)
-          Right (eraHistory, systemStart') ->
+          Right (eraHistory, mSystemStart) ->
             case slotToEpoch tipSlotNo eraHistory of
               Left e -> throwE (ShelleyQueryCmdPastHorizon e)
               Right (epochNo, _, _) -> do
                 let tipTimeResult = first toJsonPastHorizonException $
                                 fst <$> getProgress tipSlotNo eraHistory
 
-                let systemStart = getSystemStart systemStart'
-
-                nowSeconds <- toRelativeTime (SystemStart systemStart) <$> liftIO getCurrentTime
-
-                let tolerance = RelativeTime (secondsToNominalDiffTime 600)
-                let jsonSyncProgress = fmap (flip (percentage tolerance) nowSeconds) tipTimeResult
-
-                let output = encodePretty $ O.QueryTipOutput
-                      { O.chainTip = tip
-                      , O.era = anyEra
-                      , O.epoch = epochNo
-                      , O.syncProgress = jsonSyncProgress
-                      }
-
-                case mOutFile of
-                  Just (OutputFile fpath) -> liftIO $ LBS.writeFile fpath output
-                  Nothing                 -> liftIO $ LBS.putStrLn        output
-
+                case fmap getSystemStart mSystemStart of
+                  Just systemStart -> do
+                    nowSeconds <- toRelativeTime (SystemStart systemStart) <$> liftIO getCurrentTime
+
+                    let tolerance = RelativeTime (secondsToNominalDiffTime 600)
+                    let jsonSyncProgress = fmap (flip (percentage tolerance) nowSeconds) tipTimeResult
+
+                    let output = encodePretty $ O.QueryTipOutput
+                          { O.chainTip = tip
+                          , O.era = anyEra
+                          , O.epoch = epochNo
+                          , O.syncProgress = jsonSyncProgress
+                          }
+
+                    case mOutFile of
+                      Just (OutputFile fpath) -> liftIO $ LBS.writeFile fpath output
+                      Nothing                 -> liftIO $ LBS.putStrLn        output
+                  Nothing -> do
+                    let output = encodePretty $ O.QueryTipOutput
+                          { O.chainTip = tip
+                          , O.era = anyEra
+                          , O.epoch = epochNo
+                          , O.syncProgress = Left "No progress information available"
```

Comment:
Yeah, or it might be better to make this a `Maybe` type, and just omit it on output when unavailable.

Reply:
This is now a `Maybe` type.  Errors are logged to `stderr`.